MINI-PROJECTS.md

```diff
@@ -104,6 +104,314 @@ share a bound.
   those bounds is reached. Good, thoughtful test cases are essential
   here!
 
+### Instrumentation to Check Pointer Alignment
+
+Some architectures only allow load instructions / pointer dereferencing when the
+address is a multiple of the word length (i.e. 4 bytes or 8 bytes). This project
+would add a new instrumentation option for `cbmc` and `goto-instrument` that checks
+each pointer before dereference and makes sure it's correctly aligned.
+
+### Command line options to SAT solvers
+
+There are options for which SMT solver to run and there are ways of building with
+multiple SAT solvers but no command line options to pick a SAT solver. It would be
+nice if there were.
+
+César is working on this one. (?)
+
+### Combine Uninitialize Variables Analysis with Alias Analysis
+
+`goto-instrument` features an instrumentation to track cases of use-before-define, i.e.
+reading of uninitialised variables. At present, the analysis does not take into account
+aliasing information. As such, the following code will yield a spurious counterexample:
+
+``` C
+int main()
+{
+  int x;
+  int *ptr;
+  ptr=&x;
+  *ptr=42;
+  if(x < 42)
+    return 1;
+  return 0;
+}
+```
+
```

Comment:
This should be pretty straight-forward to do with `goto-analyser --vsd` and might even be a good intro to that code.